x/oracle/client/common/proof/signature.go

```diff
@@ -64,34 +62,62 @@ func recoverETHAddress(msg, sig, signer []byte) ([]byte, uint8, error) {
 }
 
 // GetSignaturesAndPrefix returns a list of TMSignature from Tendermint signed header.
-func GetSignaturesAndPrefix(info *types.SignedHeader) ([]TMSignature, error) {
+func GetSignaturesAndPrefix(info *types.SignedHeader) ([]TMSignature, CommonEncodedVotePart, error) {
 	addrs := []string{}
 	mapAddrs := map[string]TMSignature{}
-	for i, vote := range info.Commit.Signatures {
+	encodedBlockMinimal, err := protoio.MarshalDelimited(
+		&tmproto.CanonicalVote{
+			Type:   tmproto.SignedMsgType(info.Commit.Type()),
+			Height: info.Commit.Height,
+			Round:  int64(info.Commit.Round),
+			BlockID: &tmproto.CanonicalBlockID{
+				Hash: info.Commit.BlockID.Hash,
+				PartSetHeader: tmproto.CanonicalPartSetHeader{
+					Total: info.Commit.BlockID.PartSetHeader.Total,
+					Hash:  info.Commit.BlockID.PartSetHeader.Hash,
+				},
+			},
+		},
+	)
+	if err != nil {
+		return nil, CommonEncodedVotePart{}, err
+	}
+
+	lrCommon := bytes.Split(encodedBlockMinimal, info.Commit.BlockID.Hash)
```

Comment:
Can we change from split to encode the first half (Type, Height,Round) as prefix part, then encode mid part (only BlockID) and slice out blockhash part,then check is that correct and use remaining as `SignedDataSuffix` instead. To prevent rare case on split got incorrect part